### DISCUSSION TITLE
Calculating Token for GPT-4o with Images (Streaming)

### DISCUSSION THREAD

**Discussion by jmurkothintel** [microsoft/semantic-kernel#9020] [Q&A] [answered]
We need to log usage  for tracking . As the streaming api does not return usage information, so far for textual content, we have been using sharptoken to  calculate the  approximate number of tokens of the conversation . In GPT-4o as we send Base64 encoded images this approach doesn't seem to work . Would like to understand from either Microsoft or the community if there is a recommendation on  how to calculate tokens for images in the chat conversation

**Comment by moonbox3** [comment]
@jmurkothintel are you targeting .Net or Python?

**Reply by jmurkothintel** [reply]
@moonbox3  we are targetting .NET

**Reply by moonbox3** [reply, marked as answer]
Thanks for your response, @jmurkothintel. This functionality is not yet available in .Net. Here is the issue tracking it: https://github.com/microsoft/semantic-kernel/issues/9005.

**Reply by jmurkothintel** [reply]
Thanks @moonbox3  . Any rough ETA on when it is supposed to be picked up ?
